src/adagrams.js

```diff
@@ -1,7 +1,148 @@
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    const lettersDrawn = []
+    const letterDistribution = {
+      "A": 9,
+      "B": 2,
+      "C": 2,
+      "D": 4,
+      "E": 12,
+      "F": 2,
+      "G": 3,
+      "H": 2,
+      "I": 9,
+      "J": 1,
+      "K": 1,
+      "L": 4,
+      "M": 2,
+      "N": 6,
+      "O": 8,
+      "P": 2,
+      "Q": 1,
+      "R": 6,
+      "S": 4,
+      "T": 6,
+      "U": 4,
+      "V": 2,
+      "W": 2,
+      "X": 1,
+      "Y": 2,
+      "Z": 1,
+      pickRandomLetter(){
+        return Object.keys(this)[Math.floor(Math.random() * Math.floor(Object.keys(this).length - 1))];
+      },
+    }
+
+    for (let i = 0; i < 10; i++) {
+      let letter = letterDistribution.pickRandomLetter();
+
+      while(letterDistribution[letter] === 0) {
+        letter = letterDistribution.pickRandomLetter();
+      }
+
+      lettersDrawn.push(letter);
+      letterDistribution[letter] -= 1;
+    }
+   
+    return lettersDrawn;
+  },
+
+  usesAvailableLetters(input, lettersInHand){
+    let allLettersIncluded = true
+
+    for(let letter of input){
+      if (lettersInHand.includes(letter)){
+        let letterToRemove = lettersInHand.indexOf(letter);
+        lettersInHand.splice(letterToRemove, 1)
+      } else {
+        allLettersIncluded = false;
+        break;
+      }
+    }
+    return allLettersIncluded
   },
```

Comment:
You're about 50/50 for semicolons in this function.
```suggestion
    let allLettersIncluded = true;

    for(let letter of input){
      if (lettersInHand.includes(letter)){
        let letterToRemove = lettersInHand.indexOf(letter);
        lettersInHand.splice(letterToRemove, 1);
      } else {
        allLettersIncluded = false;
        break;
      }
    }
    return allLettersIncluded;
  },
```